frame/assets/src/lib.rs

```diff
@@ -747,14 +756,17 @@ pub mod pallet {
 				if asset == T::NativeAssetId::get() {
 					return <<T as Config>::NativeCurrency>::set_balance(who, amount)
 				}
+				let asset = valid_asset_id::<T>(asset).ok_or(Error::<T>::InvalidCurrency)?;
 				<<T as Config>::MultiCurrency>::set_balance(asset, who, amount)
 			}
 
 			fn set_total_issuance(asset: Self::AssetId, amount: Self::Balance) {
 				if asset == T::NativeAssetId::get() {
 					return <<T as Config>::NativeCurrency>::set_total_issuance(amount)
 				}
-				<<T as Config>::MultiCurrency>::set_total_issuance(asset, amount)
+				if let Some(asset) = valid_asset_id::<T>(asset) {
+					<<T as Config>::MultiCurrency>::set_total_issuance(asset, amount)
+				}
```

Comment:
What if invalid?

Reply:
function signature does not return anything so if invalid do nothing.